### PR TITLE
keras 2 merge layers

### PR DESCRIPTION
**Pull request by maxpumperla**
Simply adding explicit keras 2 merge layers, e.g. `Add` vs. `Merge(mode='add')`.

### CODE REVIEW

File: deeplearning4j-modelimport/src/main/java/org/deeplearning4j/nn/modelimport/keras/utils/KerasLayerUtils.java [hunk 1]
```diff
@@ -225,6 +226,24 @@ public static KerasLayer getKerasLayerFromConfig(Map<String, Object> layerConfig
...
+                layerClassName.equals(conf.getLAYER_CLASS_NAME_FUNCTIONAL_SUBTRACT())) {
+            layer = new KerasMerge(layerConfig, ElementWiseVertex.Op.Subtract, enforceTrainingConfig);
+        } else if (layerClassName.equals(conf.getLAYER_CLASS_NAME_AVERAGE()) ||
+                layerClassName.equals(conf.getLAYER_CLASS_NAME_FUNCTIONAL_AVERAGE())) {
+            layer = new KerasMerge(layerConfig, ElementWiseVertex.Op.Average, enforceTrainingConfig);
+        } else if (layerClassName.equals(conf.getLAYER_CLASS_NAME_MULTIPLY()) ||
+                layerClassName.equals(conf.getLAYER_CLASS_NAME_FUNCTIONAL_MULTIPLY())) {
+            layer = new KerasMerge(layerConfig, ElementWiseVertex.Op.Product, enforceTrainingConfig);
+        } else if (layerClassName.equals(conf.getLAYER_CLASS_NAME_ADD()) ||
+                layerClassName.equals(conf.getLAYER_CLASS_NAME_ADD())) {
```
Comment (AlexDBlack) [inline deeplearning4j-modelimport/src/main/java/org/deeplearning4j/nn/modelimport/keras/utils/KerasLayerUtils.java]: Isn't the add case already handled earlier?

File: deeplearning4j-modelimport/src/main/java/org/deeplearning4j/nn/modelimport/keras/utils/KerasLayerUtils.java [hunk 2]
```diff
@@ -225,6 +226,24 @@ public static KerasLayer getKerasLayerFromConfig(Map<String, Object> layerConfig
...
+            layer = new KerasMerge(layerConfig, ElementWiseVertex.Op.Add, enforceTrainingConfig);
+        } else if (layerClassName.equals(conf.getLAYER_CLASS_NAME_DOT()) ||
+                layerClassName.equals(conf.getLAYER_CLASS_NAME_FUNCTIONAL_DOT())) {
+            layer = new KerasMerge(layerConfig, null, enforceTrainingConfig);
```
Comment (AlexDBlack) [inline deeplearning4j-modelimport/src/main/java/org/deeplearning4j/nn/modelimport/keras/utils/KerasLayerUtils.java]: If mergeMode is null, wan't this use MergeVertex, and not implement dot?

File: deeplearning4j-modelimport/src/main/java/org/deeplearning4j/nn/modelimport/keras/utils/KerasLayerUtils.java (hunk 1 above)
Comment (maxpumperla) [reply deeplearning4j-modelimport/src/main/java/org/deeplearning4j/nn/modelimport/keras/utils/KerasLayerUtils.java]: yes, thanks!

File: deeplearning4j-modelimport/src/main/java/org/deeplearning4j/nn/modelimport/keras/utils/KerasLayerUtils.java (hunk 2 above)
Comment (maxpumperla) [reply deeplearning4j-modelimport/src/main/java/org/deeplearning4j/nn/modelimport/keras/utils/KerasLayerUtils.java]: yep, that should be concatenate... :/

**Pull request merged**